src/Umbraco.Infrastructure/Extensions/EmailMessageExtensions.cs

```diff
@@ -58,6 +55,58 @@ public static MimeMessage ToMimeMessage(this EmailMessage mailMessage, string co
             return messageToSend;
         }
 
+        public static NotificationEmailModel ToNotificationEmail(this EmailMessage emailMessage,
+            string configuredFromAddress)
+        {
+            var fromEmail = string.IsNullOrEmpty(emailMessage.From) ? configuredFromAddress : emailMessage.From;
+
+            NotificationEmailAddress from = ToNotificationAddress(fromEmail);
+
+            return new NotificationEmailModel(from,
+                GetNotificationAddresses(emailMessage.To),
+                GetNotificationAddresses(emailMessage.Cc),
+                GetNotificationAddresses(emailMessage.Bcc),
+                GetNotificationAddresses(emailMessage.ReplyTo),
+                emailMessage.Subject,
+                emailMessage.Body,
+                emailMessage.Attachments,
+                emailMessage.IsBodyHtml);
+        }
+
+        private static NotificationEmailAddress ToNotificationAddress(string address)
+        {
+            if (InternetAddress.TryParse(address, out InternetAddress internetAddress))
+            {
+                if (internetAddress is MailboxAddress mailboxAddress)
+                {
+                    return new NotificationEmailAddress(mailboxAddress.Address, internetAddress.Name);
+                }
+            }
+
+            return null;
+        }
+
+        private static IEnumerable<NotificationEmailAddress> GetNotificationAddresses(IEnumerable<string> addresses)
+        {
+            if (addresses is null)
+            {
+                return null;
+            }
+
+            var notificationAddresses = new List<NotificationEmailAddress>();
+
+            foreach (var address in addresses)
+            {
+                NotificationEmailAddress notificationAddress = ToNotificationAddress(address);
+                if (notificationAddress is not null)
+                {
+                    notificationAddresses.Add(notificationAddress);
+                }
+            }
+
+            return notificationAddresses;
+        }
+
         private static void AddAddresses(MimeMessage message, string[] addresses, Func<MimeMessage, InternetAddressList> addressListGetter, bool throwIfNoneValid = false)
```

Comment:
Just to make the file easier to read top to bottom, maybe move this helper method up to below the public method that uses it.  Or if you prefer public methods at the top, make this the first private helper method in the class.

Reply:
Good point, I don't think it's too big an issue, so I moved the helper method up below the public method.

Reply:
No, no big deal - just think it's nice when you read a class from to to bottom that methods are in an order reflecting how they are called.  Sounds good.

Reply:
Oh I agree, I was referring to the mixing of public/private methods 😄